src/js/utils/FormUtil.js

```diff
@@ -0,0 +1,89 @@
+function cloneDefinition(array) {
+  return array.map(function (object) {
+    if (Array.isArray(object)) {
+      return object.slice();
+    }
+
+    if (typeof object === 'object') {
+      return Object.assign({}, object);
+    }
+
+    return object;
+  });
+}
+
+function getProp(key) {
+  return key.split('[')[0];
+}
+
+function getPropIndex(key) {
+  let value = key.split('[');
+  return parseInt(value[1].split(']')[0])
```

Comment:
The two lines above could better be solved with a regular expression to be clearer in what is happening.
```JS
parseInt(key.match(\[(\d*)\])[1], 10)
```
And to be a bit more error safe there is also this option
```JS
parseInt(key.match(\[\d*\]).join().replace(/\[|\]/, ''), 10)
```